src/main/java/core/basesyntax/handler/impl/SupplyTransactionHandler.java

```diff
@@ -0,0 +1,12 @@
+package core.basesyntax.handler.impl;
+
+import core.basesyntax.handler.TransactionHandler;
+import java.util.Map;
+
+public class SupplyTransactionHandler implements TransactionHandler {
+    @Override
+    public void handleTransaction(Map<String, Integer> mapFruitQuantity, String key, int value) {
+        mapFruitQuantity.put(key, mapFruitQuantity.getOrDefault(key, 0) + value);
+    }
+
+}
```

Comment:
This class seems to be handling the 'Supply' transaction correctly by adding the supplied quantity to the current stock. However, please ensure that this class is part of a package that groups similar handlers together, following the checklist requirement to keep all logic not in a single package but rather organized in corresponding packages. For example, this could be part of a 'strategy' package if you are applying the Strategy pattern.